infrastructure/lib/kit-infrastructure.ts

```diff
@@ -34,7 +35,10 @@ export class KITInfrastructure extends Stack {
 
     // The IAM role that will be used by EKS
     const clusterRole = new iam.Role(this, 'ClusterRole', {
-      assumedBy: new iam.ServicePrincipal('eks.amazonaws.com'),
+      assumedBy: new iam.CompositePrincipal(
+        new iam.ArnPrincipal(`arn:aws:iam::${cdk.Aws.ACCOUNT_ID}:root`),
```

Comment:
Have you tested this part? Are you sure we should use `root` instead of `admin`?

Reply:
Yes, I tested this part. It gives the Admin to assume this role to access this cluster. Given its admin role, should be allowed to assume.